src/fastmcp/server/auth/cimd.py

```
@@ -0,0 +1,455 @@
+            return False
+        try:
+            parsed = urlparse(client_id)
+            return parsed.scheme == "https" and parsed.path not in ("", "/")
+        except Exception:
+            return False
+
+    def _is_private_ip(self, hostname: str) -> bool:
+        """Check if hostname resolves to a private/loopback IP (SSRF protection)."""
+        try:
+            ip = ipaddress.ip_address(hostname)
+            return ip.is_private or ip.is_loopback or ip.is_reserved
```

Ruff (0.14.11)

Consider moving this statement to an `else` block

(TRY300)